### PR TITLE
Python 3: replace deprecated Assertregexpmatches use for Py3

### PR DESCRIPTION
**Pull request by ziransun**

### CODE REVIEW

File: tools/wptserve/tests/functional/test_pipes.py [hunk 1]
```diff
@@ -120,7 +121,11 @@ def test_sub_url_base_via_filename_with_query(self):
 
     def test_sub_uuid(self):
         resp = self.request("/sub_uuid.sub.txt")
-        self.assertRegexpMatches(resp.read().rstrip(), b"Before [a-f0-9-]+ After")
```
Comment (stephenmcgruer) [inline tools/wptserve/tests/functional/test_pipes.py:123]: Shouldn't this just use https://six.readthedocs.io/#six.assertRegex then?

Comment (ziransun) [reply tools/wptserve/tests/functional/test_pipes.py:123]: Yes, I had thought about it. Then I thought maybe for consistence of calling "self.*" through the class?

Comment (ziransun) [reply tools/wptserve/tests/functional/test_pipes.py:123]: Updated :)

File: tools/wptserve/tests/functional/test_pipes.py [hunk 2]
```diff
@@ -43,11 +44,11 @@ def test_semicolon(self):
         self.assertEqual(resp.info()["Refresh"], "3;url=http://example.com")
 
     def test_escape_comma(self):
-        resp = self.request("/document.txt", query="pipe=header(Expires,Thu\,%2014%20Aug%201986%2018:00:00%20GMT)")
+        resp = self.request("/document.txt", query=r"pipe=header(Expires,Thu\,%2014%20Aug%201986%2018:00:00%20GMT)")
```
Comment (stephenmcgruer) [inline tools/wptserve/tests/functional/test_pipes.py:47]: Sorry, I missed these changes before. Why are these changes required?

Comment (ziransun) [reply tools/wptserve/tests/functional/test_pipes.py:47]: This is because we have `\` in the string. Python 3 interprets string literals as Unicode strings. In this case Py3 gives warning like `DeprecationWarning: invalid escape sequence \`. Using a raw string here is to make sure that `\` is interpreted as a plain `\` rather than escape sequences. It is also recommended in [re](https://docs.python.org/3/library/re.html) to use raw string for regular expressions.

Comment (stephenmcgruer) [reply tools/wptserve/tests/functional/test_pipes.py:47]: Where is the regexp in this test? This test appears to send a request to the wptserve server started by the `TestUsingServer` class (in `tools/wptserve/tests/functional/base.py`), and asserts that the return value has the right 'Expires' value (subbed in via a `pipe`). More specifically, from what I can see the `query` parameter is:

1. Passed into `request`
1. Then passed into `self.abs_url`
1. Then passed into `urlunsplit` (aka `six.moves.urllib.parse.urlunsplit`)
1. The result of `urlunsplit` is then passed to the `Request` class constructor
1. The `Request` class constructor passed it to `BaseRequest` (aka `six.moves.urllib.request.Request`).

Which part of this uses a regexp? (Or did I miss something).

It seems like adding `r` here **might** be covering up something that may be a bug, or it may be fine, and I'm not confident currently which it is! :)

Comment (ziransun) [reply tools/wptserve/tests/functional/test_pipes.py:47]: Ah sorry! There is no regexp call here. I have messed it up with the change of "assertRegex".

Comment (ziransun) [reply tools/wptserve/tests/functional/test_pipes.py:47]: > This is because we have `\` in the string. Python 3 interprets string literals as Unicode strings. In this case Py3 gives warning like `DeprecationWarning: invalid escape sequence \`. Using a raw string here is to make sure that `\` is interpreted as a plain `\` rather than escape sequences. 

This is the reason we use raw string here.

Comment (stephenmcgruer) [reply tools/wptserve/tests/functional/test_pipes.py:47]: Ok. I think I understand; the `regex` part wasn't actually relevant, but this is still using an unescaped backslash in a string which is a DeprecationWarning in Py3+. The reason being that if our comma was actually a valid escape sequence, we would end up with a different string:

```
>>> x = "\ndsf"
>>> y = r"\ndsf"
>>> x
'\ndsf'
>>> y
'\\ndsf'
```

Ok, this LGTM.

**Review by stephenmcgruer — APPROVED** [review]

**Comment by stephenmcgruer** [comment]
CI failures were https://github.com/web-platform-tests/wpt/issues/26131, for the record
